libsam.c

```diff
@@ -528,18 +527,22 @@ int sam_put_grp_members_sid(struct hive *hdesc, int grp, struct sid_array *sarra
     sidptr = &cd->data[mofs];
 
     for (i = 0; sarray[i].sidptr; i++) {
-      if (gverbose) printf("  copying : %d len %x, at %x\n",i,sarray[i].len, sidptr);
-      str = sam_sid_to_string(sarray[i].sidptr);
-      if (gverbose) printf("  Member # %d = <%s>\n", i, str);
-      FREE(str);      
+      if (gverbose)
+      {
+          printf("  copying : %d len %x, at %x\n",i,sarray[i].len, (unsigned int)sidptr);
+          str = sam_sid_to_string(sarray[i].sidptr);
+          printf("  Member # %d = <%s>\n", i, str);
+          FREE(str);
+      }
+
```

Comment:
gverbose should only be used to turn on or off the printf statement.
Else if gverbose is false then your code does not do anything useful.